src/aead/mod.rs

```diff
@@ -334,12 +342,14 @@ const TAG_LEN: usize = poly1305::TAG_LEN;
 const NONCE_LEN: usize = 96 / 8;
 
 
-/// |GFp_chacha_20| uses a 32-bit block counter, so we disallow individual
-/// operations that work on more than 256GB at a time, for all AEADs.
-fn check_per_nonce_max_bytes(in_out_len: usize)
+fn check_per_nonce_max_bytes(alg: &Algorithm, in_out_len: usize)
                              -> Result<(), error::Unspecified> {
-    if polyfill::u64_from_usize(in_out_len) >= (1u64 << 32) * 64 - 64 {
+    if polyfill::u64_from_usize(in_out_len) > alg.max_input_len {
```

Comment:
Note the change from `>=` to `>` here. I'm at a loss as to why I was using `>=` before.